### ISSUE TITLE
Errors within genReqId crash the entire server on request

### ISSUE THREAD

**Issue by NOT-HAL9000** [fastify/fastify#3872]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

3.29.0

### Node.js version

16.15

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

-

### Description

I'm not sure if this is the expected behaviour, but if there is an error within ```genReqId```, the entire Fastify server crashes and no more requests can be processed.

### Steps to Reproduce

Init Fastify with the following options and curl the server.
```javascript
const fastify = Fastify({
  logger: true
  genReqId: (req) => {  throw new Error('lib failed to generate id') },
})
```

### Expected Behavior

I was expecting it to fail gracefully. Given that the request id isn't vital to the application, a failure in the generation function shouldn't crash the entire server. 

I would expect something like
- The error should be logged (```fastify.log.error(...)```)
- Fastify should fallback to using its native request-id generator
- The request is processed

**Comment by mcollina** [comment]
It seems about right. This method should not error, we should update the docs.

**Comment by Fdawgs** [comment]
Closed by #3898
